regression: reflect.Set: value of type stdlib._testing_TB is not assignable to type testing.TB

### The following program `sample.go` triggers an unexpected result

```go
https://github.com/traefik/plugin-simplecache
```


### Expected result

```console
No panic
```


### Got

```console
$ make yaegi_test 
yaegi test -v .
=== RUN   TestFileCache
run.go:1015: file_test.go:16:23: genInterfaceWrapper error, no method private
file_test.go:35:14: panic
--- FAIL: TestFileCache (0.00s)
panic: reflect.Set: value of type stdlib._testing_TB is not assignable to type testing.TB [recovered]
        panic: reflect.Set: value of type stdlib._testing_TB is not assignable to type testing.TB [recovered]
        panic: reflect.Set: value of type stdlib._testing_TB is not assignable to type testing.TB

goroutine 10 [running]:
testing.tRunner.func1.2(0xe24100, 0xc00099fac0)
        /home/ldez/.gvm/gos/go1.16.6/src/testing/testing.go:1143 +0x332
testing.tRunner.func1(0xc000216480)
        /home/ldez/.gvm/gos/go1.16.6/src/testing/testing.go:1146 +0x4b6
panic(0xe24100, 0xc00099fac0)
        /home/ldez/.gvm/gos/go1.16.6/src/runtime/panic.go:965 +0x1b9
github.com/traefik/yaegi/interp.runCfg.func1(0xc000982580, 0xc000900300, 0xc000825c78)
        /home/ldez/sources/go/src/github.com/traefik/yaegi/interp/run.go:185 +0x253
panic(0xe24100, 0xc00099fac0)
        /home/ldez/.gvm/gos/go1.16.6/src/runtime/panic.go:965 +0x1b9
reflect.Value.assignTo(0xf4b880, 0xc0001d23c0, 0x199, 0xf63fa7, 0xb, 0xf05240, 0xc00099f810, 0x0, 0xe24100, 0x19)
        /home/ldez/.gvm/gos/go1.16.6/src/reflect/value.go:2451 +0x3f7
reflect.Value.Set(0xf05240, 0xc00099f810, 0x194, 0xf4b880, 0xc0001d23c0, 0x199)
        /home/ldez/.gvm/gos/go1.16.6/src/reflect/value.go:1564 +0xbd
github.com/traefik/yaegi/interp.call.func6(0xc000982580, 0x16)
        /home/ldez/sources/go/src/github.com/traefik/yaegi/interp/run.go:1264 +0xafd
github.com/traefik/yaegi/interp.runCfg(0xc000900300, 0xc000982580)
        /home/ldez/sources/go/src/github.com/traefik/yaegi/interp/run.go:191 +0x87
github.com/traefik/yaegi/interp.genFunctionWrapper.func2.1(0xc000994330, 0x1, 0x1, 0xc000994330, 0x153f478, 0x11b418a)
        /home/ldez/sources/go/src/github.com/traefik/yaegi/interp/run.go:952 +0x3e8
testing.tRunner(0xc000216480, 0xc000253860)
        /home/ldez/.gvm/gos/go1.16.6/src/testing/testing.go:1193 +0xef
created by testing.(*T).Run
        /home/ldez/.gvm/gos/go1.16.6/src/testing/testing.go:1238 +0x2b3
make: *** [Makefile:14: yaegi_test] Error 2
```


### Yaegi Version

v0.9.20

### Additional Notes

Everything works with v0.9.19

--- a/_test/interface52.go
+++ b/_test/interface52.go
@@ -1,6 +1,7 @@
 package main
 
 import (
+	"log"
 	"os"
 	"strings"
 	"testing"
@@ -15,7 +16,7 @@ func main() {
 		println("FAIL")
 		return
 	}
-	println("tmpdir:", tmpdir, "testing tmpdir:", tb.TempDir())
+	log.Println("tmpdir:", tmpdir, "testing tmpdir:", tb.TempDir())
 	if !strings.HasPrefix(tb.TempDir(), tmpdir) {
 		println("FAIL")
 		return

--- a/_test/interface52.go
+++ b/_test/interface52.go
@@ -1,7 +1,7 @@
 package main
 
 import (
-	"log"
+	"fmt"
 	"os"
 	"strings"
 	"testing"
@@ -16,7 +16,7 @@ func main() {
 		println("FAIL")
 		return
 	}
-	log.Println("tmpdir:", tmpdir, "testing tmpdir:", tb.TempDir())
+	fmt.Fprintln(os.Stdout, "tmpdir:", tmpdir, "testing tmpdir:", tb.TempDir())
 	if !strings.HasPrefix(tb.TempDir(), tmpdir) {
 		println("FAIL")
 		return

--- a/_test/interface52.go
+++ b/_test/interface52.go
@@ -15,6 +15,7 @@ func main() {
 		println("FAIL")
 		return
 	}
+	println("tmpdir:", tmpdir, "testing tmpdir:", tb.TempDir())
 	if !strings.HasPrefix(tb.TempDir(), tmpdir) {
 		println("FAIL")
 		return

--- a/_test/interface52.go
+++ b/_test/interface52.go
@@ -1,9 +1,6 @@
 package main
 
 import (
-	"fmt"
-	"os"
-	"strings"
 	"testing"
 )
 
@@ -11,13 +8,14 @@ func main() {
 	t := testing.T{}
 	var tb testing.TB
 	tb = &t
-	tmpdir := os.Getenv("TMPDIR")
-	if tmpdir == "" {
-		println("FAIL")
-		return
-	}
-	fmt.Fprintln(os.Stdout, "tmpdir:", tmpdir, "testing tmpdir:", tb.TempDir())
-	if !strings.HasPrefix(tb.TempDir(), tmpdir) {
+	//tmpdir := os.Getenv("TMPDIR")
+	//if tmpdir == "" {
+	//	println("FAIL")
+	//	return
+	//}
+	// fmt.Fprintln(os.Stdout, "tmpdir:", tmpdir, "testing tmpdir:", tb.TempDir())
+	// if !strings.HasPrefix(tb.TempDir(), tmpdir) {
+	if tb.TempDir() == "" {
 		println("FAIL")
 		return
 	}

--- a/_test/interface52.go
+++ b/_test/interface52.go
@@ -1,20 +1,11 @@
 package main
 
-import (
-	"testing"
-)
+import "testing"
 
 func main() {
 	t := testing.T{}
 	var tb testing.TB
 	tb = &t
-	//tmpdir := os.Getenv("TMPDIR")
-	//if tmpdir == "" {
-	//	println("FAIL")
-	//	return
-	//}
-	// fmt.Fprintln(os.Stdout, "tmpdir:", tmpdir, "testing tmpdir:", tb.TempDir())
-	// if !strings.HasPrefix(tb.TempDir(), tmpdir) {
 	if tb.TempDir() == "" {
 		println("FAIL")
 		return

--- a/_test/interface52.go
+++ b/_test/interface52.go
@@ -1,0 +1,26 @@
+package main
+
+import (
+	"os"
+	"strings"
+	"testing"
+)
+
+func main() {
+	t := testing.T{}
+	var tb testing.TB
+	tb = &t
+	tmpdir := os.Getenv("TMPDIR")
+	if tmpdir == "" {
+		println("FAIL")
+		return
+	}
+	if !strings.HasPrefix(tb.TempDir(), tmpdir) {
+		println("FAIL")
+		return
+	}
+	println("PASS")
+}
+
+// Output:
+// PASS

--- a/interp/run.go
+++ b/interp/run.go
@@ -971,7 +971,8 @@ func genInterfaceWrapper(n *node, typ reflect.Type) func(*frame) reflect.Value {
 	if typ == nil || typ.Kind() != reflect.Interface || typ.NumMethod() == 0 || n.typ.cat == valueT {
 		return value
 	}
-	if nt := n.typ.TypeOf(); nt != nil && nt.Kind() == reflect.Interface {
+	nt := n.typ.frameType()
+	if nt != nil && nt.Implements(typ) {
 		return value
 	}
 	mn := typ.NumMethod()
@@ -990,6 +991,9 @@ func genInterfaceWrapper(n *node, typ reflect.Type) func(*frame) reflect.Value {
 
 	return func(f *frame) reflect.Value {
 		v := value(f)
+		if v.Type().Implements(typ) {
+			return v
+		}
 		vv := v
 		switch v.Kind() {
 		case reflect.Chan, reflect.Func, reflect.Interface, reflect.Map, reflect.Ptr, reflect.Slice:

